StoreBroker/StoreIngestionPackageApi.ps1

```
@@ -485,10 +485,19 @@ function Get-PackagesToKeep
 
     foreach ($entry in $uniquePackageTypeMapping.Keys)
     {
-        $null = $uniquePackageTypeMapping[$entry] | Sort-Object version
-        $null = [array]::Reverse($uniquePackageTypeMapping[$entry])
+        # Sort-Object change the output into a single object if the length of the input array is 1
+        if ($uniquePackageTypeMapping[$entry].Count -gt 1)
+        {
+            $sortedPackageInfo = $uniquePackageTypeMapping[$entry] | Sort-Object version -Descending
+        }
+        else 
+        {
+            $sortedPackageInfo = $uniquePackageTypeMapping[$entry]
+        }
+
         # We map each package type with the versions of the packages, and for each package type, the maximum number of packages to keep is defined by RedendantPackagesToKeep
-        foreach ($pkgObj in $uniquePackageTypeMapping[$entry][0..($RedundantPackagesToKeep - 1)])
+        # We are not expecting the number of packages to keep is less than or equal to zero.
```

A comment won't make this so.  If you have a restriction that this method can only run with a value of 1 or greater, then you should have that restriction declared on the parameter itself:
`[ValidateRange(1, 90)]`
You can use an arbitrarily high max value for the range if you want to let the code silently fallback to `$MaxPackagesPerGroup`, or just explicitly declare 25 as the max in the range and then remove your constant declaration and fallback logic in the method.
The thing that you _shouldn't_ do is just have a comment saying that you don't expect a certain type of value and not protect against it...especially because the way that this is written, if a user _did_ provide `0`, this would end up setting up to `2` packages to be retained (`-1` causes it to start grabbing from the end of the array), which is very much

---

For the arbitrary large value, is it fine to use something like $([uint32]::MaxValue) ?